docs/API Documentation.md

```diff
@@ -0,0 +1,25 @@
+# API documentation
+This file provides a list of endpoints for the backend.
+
+## register
+* Endpoint: `/register`
+* Body parameters:
+    * `username`: string of 1-100 characters. Must be unique.
+    * `email`: string of 5-255 characters. Must be unique.
+    * `firstname`: string of 1-100 characters.
+    * `lastname`: string of up 1-100 characters.
+    * `password`: string of up to 5-255 characters. Currently, there are not any kind of complexity check, excluding length restrictions.
```

Comment:
whoops, should be "checks," will fix later.

Reply:
also didn't note that this is a POST. also TODO: add more examples operations with example responses